infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/aws/StackCreationValidator.java

```diff
@@ -0,0 +1,133 @@
+import com.amazonaws.services.cloudformation.model.StackStatus;
+import org.awaitility.Awaitility;
+import org.awaitility.core.ConditionTimeoutException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.testgrid.common.exception.TestGridInfrastructureException;
+import org.wso2.testgrid.common.util.StringUtil;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Validates the Stack creation in AWS.
+ */
+public class StackCreationValidator {
+
+    private static final Logger logger = LoggerFactory.getLogger(StackCreationValidator.class);
+
+    /**
+     * Periodically checks for the status of stack creation until a defined timeout.
+     *
+     * @param timeout         Time to wait upon the desired response.
+     * @param timeoutTimeUnit TimeUnit of timeout value.
+     * @param pollInterval    Time interval to perform polling.
+     * @param pollTimeUnit    TimeUnit of pollInterval.
+     */
+    public void waitForStack(int timeout, TimeUnit timeoutTimeUnit, int pollInterval,
+                                  TimeUnit pollTimeUnit, String stackName, AmazonCloudFormation cloudFormation)
+            throws ConditionTimeoutException {
+        Awaitility.with().pollInterval(pollInterval, pollTimeUnit).await().
+                atMost(timeout, timeoutTimeUnit).until(isStackCreationSuccessful(stackName, cloudFormation));
+    }
+
+    /**
+     * Checks if a defined stack gets to CREATE_COMPLETE state.
+     *
+     * @param stackName Name of the stack to wait
+     * @param cloudFormation Amazon cloud formation
+     * @return true for CREATE_COMPLETE state and false for failed states.
+     */
+    private Callable<Boolean> isStackCreationSuccessful(String stackName, AmazonCloudFormation cloudFormation) {
+        return new AWSCallable(stackName, cloudFormation);
+    }
+
+    /**
+     * Inner class for the callable implementation used by awaitility to determine the
+     * condition of the stack.
+     */
+    private static class AWSCallable implements Callable<Boolean> {
+        private String stackName;
+        private AmazonCloudFormation cloudFormation;
+
+        /**
+         * Constructs the AWSCallable object with stackname and cloudformation.
+         *
+         * @param stackName Name of the stack to wait
+         * @param cloudFormation Amazon cloud formation
+         */
+        AWSCallable(String stackName, AmazonCloudFormation cloudFormation) {
+            this.stackName = stackName;
+            this.cloudFormation = cloudFormation;
+        }
+
+        @Override
+        public Boolean call() throws Exception {
+            //Stack details
+            DescribeStacksRequest describeStacksRequest = new DescribeStacksRequest().withStackName(stackName);
+            DescribeStacksResult describeStacksResult = cloudFormation.describeStacks(describeStacksRequest);
+            for (int i = 0; i < describeStacksResult.getStacks().size(); i++) {
```

Comment:
We should print the logs for the stack we created in the current test-run. It can be identified by the stackId. So, the for-loop in not necessary.

Reply:
We have to periodically check the status using `cloudFormation.describeStacks(describeStacksRequest)` to get the latest status which returns an array.